ERCS/eip-7281.md

```diff
@@ -0,0 +1,362 @@
+   */
+
+  error IXERC20Lockbox_Native();
+
+  /**
+   * @notice Reverts when a user tries to withdraw and the call fails
+   */
+
+  error IXERC-20Lockbox_WithdrawFailed();
+
+  /**
+   * @notice Deposit ERC-20 tokens into the lockbox
+   *
+   * @param _amount The amount of tokens to deposit
+   */
+
+  function deposit(uint256 _amount) external;
+
+  /**
+   * @notice Withdraw ERC-20 tokens from the lockbox
+   *
+   * @param _amount The amount of tokens to withdraw
+   */
+
+  function withdraw(uint256 _amount) external;
+}
+```
+
+Lockboxes SHOULD additionally implement the following alternative `deposit` function for native (non-ERC-20) assets.
+
+```ts
+/**
+   * @notice Deposit native assets (e.g. ETH) into the lockbox
+   */
+
+  function deposit() external payable;
+```
+
+
+## Rationale
+
+The proposed standard attempts to satisfy the following conditions for bridged tokens regardless of where and how they are bridged:
+
+1. Tokens received at the destination should be the canonical tokens.
+2. Bridges should not need to bootstrap liquidity for each new chain and asset.
+3. Cross-domain interactions involving tokens should be slippage-free, simplifying composability.
+4. Token issuers should own decision-making around which bridges to support and be able to parametrize risk for each. They should not be locked into only supporting a single bridge.
+
+### Fungibility
+
+The core problem that this standard tackles is the **fungibility** of tokens bridged across domains against the “canonical” token on that domain. (Note: which token is “canonical” is dictated by the token issuer and is sometimes, but not always, the token minted by a given domain’s enshrined bridge - e.g. a rollup bridge).
+
+![](../assets/eip-draft_bridged_tokens/fragmentation.png)
+
+It is not enough to simply allow multiple bridges to mint the same representation on a remote domain if liquidity is locked into bridges on the home domain. To illustrate this, consider an example where two bridges control minting rights of canonical USDT on an L2:
+
+- Alice bridges 100 USDT from L1→L2 through bridge 1. The underlying L1 USDT tokens is locked in bridge 1 and 100 USDT is minted on L2.
+- Bob bridges 100 USDT from L1→L2 through bridge 2. Similar to the above case, the underlying L1 USDT tokens are locked in bridge 2 and 100 USDT is minted on L2.
+- Suppose Alice pays Bob her 100 USDT. Bob now has 200 USDT.
+- Bob attempts to bridge the 200 USDT from L2→L1 through bridge 2. This transaction fails, because bridge 2 only has 100 USDT custodied that it can give to Bob.
+
+The core property that this example illustrates is that locking tokens across multiple bridges on the token’s “home” domain makes it impossible to have fungibility without impeding user experience on remote domains. Minting or burning the token can solve this problem, but not all ERC-20 tokens implement a configurable mint/burn interface. 
```

Comment:
This section doesn't seem to be explaining any particular choice(s) made within the proposal. Instead it seems like it's justifying the proposal as a whole? If so, this belongs in the Motivation section.

The Motivation should answer questions like "why do we need sovereign bridge tokens that are fungible?", while the Rationale section should answer questions like "why did we choose to include the `setLockbox` function?"